common/assertions/events.py

```diff
@@ -0,0 +1,92 @@
+
+import json
+import pprint
+
+
+def assert_event_matches(expected, actual, strict=False):
+    """
+    Compare two event dictionaries.
+
+    Fail if any discrepancies exist, and output the list of all discrepancies. The intent is to produce clearer
+    error messages than "{ some massive dict } != { some other massive dict }", instead enumerating the keys that
+    differ. Produces period separated "paths" to keys in the output, so "context.foo" refers to the following
+    structure:
+
+        {
+            'context': {
+                'foo': 'bar'  # this key, value pair
+            }
+        }
+
+    By default, it only asserts that all fields specified in the first event also exist and have the same value in
+    the second event. If the `strict` parameter is passed in it will also ensure that *only* the fields in the first
+    event exist in the second event. For example::
+
+        expected = {
+            'a': 'b'
+        }
+
+        actual = {
+            'a': 'b',
+            'c': 'd'
+        }
+
+        assert_event_matches(expected, actual, strict=False)  # This will not raise an AssertionError
+        assert_event_matches(expected, actual, strict=True)   # This *will* raise an AssertionError
+    """
+    differences = check_event_match(expected, actual, strict)
+    if len(differences) > 0:
+        debug_info = [
+            'Expected:',
+            get_pretty_event_string(expected),
+            'Actual:',
+            get_pretty_event_string(actual),
+        ]
+        raise AssertionError('Unexpected event differences found:\n' + '\n'.join(differences + debug_info))
+
+
+def check_event_match(expected, actual, strict=False, path=None):
+    if path is None:
+        path = []
+    differences = []
+
+    # Some events store their payload in a JSON string instead of a dict. Comparing these strings can be problematic
+    # since the keys may be in different orders, so we parse the string here if we were expecting a dict.
+    if not strict and path == ['event'] and isinstance(expected, dict) and isinstance(actual, basestring):
+        actual = json.loads(actual)
```

Comment:
To amplify on the previous comment, the assert_event_matches method could contain the conversion of the 'actual' string to a dict.  But it can also do other event-specific work.  By default it could check the entirety of the event against expected-content non-strictly, but it could also specifically check the payload to assert that it match strictly.  It could provide override flags that would on the one hand check the entire event strictly, and on the other check the payload non-strictly.  The former would be good for the occasional exact test.  I'm not sure if the latter were needed, but it could be added quickly if it were needed.  (E.g. 'strict' version 'payload_strict', with defaults of False and True respectively.)